ocs_ci/ocs/scale_noobaa_lib.py

```diff
@@ -154,3 +157,87 @@ def get_hpa_utilization(namespace):
         value_list = [item for elem in value for item in elem]
         hpa_cpu_utilization = int(value_list[0])
     return hpa_cpu_utilization
+
+
+def noobaa_running_node_restart(pod_name):
+    """
+    Function to restart node which has noobaa pod's running
+
+    Args:
+        pod_name (str): Name of noobaa pod
+
+    """
+    node_name = get_node_where_noobaa_pod_running(pod_name=pod_name)
+    log.info(f"{pod_name} is running on: {node_name}")
+    log.info(f"Restating node: {node_name}")
+    factory = platform_nodes.PlatformNodesFactory()
+    nodes = factory.get_nodes_platform()
+    ocp_nodes = get_node_objs(node_names=node_name)
+    nodes.restart_nodes_by_stop_and_start(nodes=ocp_nodes, force=True)
+
+    # Validate nodes are up and running
+    wait_for_nodes_status()
+    ceph_health_check(tries=30, delay=60)
+
+    # Validate noobaa pod is re-spinned and in running state
+    pod_obj = pod.get_pod_obj(
+        (
+            get_pod_name_by_pattern(
+                pattern=pod_name, namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+            )
+        )[0],
+        namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+    )
+    helpers.wait_for_resource_state(pod_obj, constants.STATUS_RUNNING, timeout=180)
+
+
+def get_node_where_noobaa_pod_running(pod_name):
```

Comment:
why not simply use `get_pod_node` from pod.py?

Reply:
@tiffanyn108 Did you tried this ?